src/imap/scan_folders.rs

```diff
@@ -0,0 +1,102 @@
+use std::time::{Instant, SystemTime};
+
+use crate::{config::Config, context::Context};
+use anyhow::Context as _;
+
+use crate::error::Result;
+use crate::imap::Imap;
+use async_std::prelude::*;
+
+use super::{get_folder_meaning, get_folder_meaning_by_name, FolderMeaning};
+
+impl Imap {
+    pub async fn scan_folders(&mut self, context: &Context) -> Result<()> {
+        use crate::config::Config::*;
+
+        // First of all, debounce to once per minute:
+        let mut last_scan = context.last_full_folder_scan.lock().await;
+        if let Some(last_scan) = *last_scan {
+            if last_scan.elapsed().as_secs() < 60 {
+                // For the first day after installation, we only debounce to 2s:
+                let configure = context.get_config(Config::ConfiguredTimestamp).await;
+                let configure = configure.context("scan_folders: not configured")?;
+                let configure: SystemTime = serde_json::from_str(&configure)?;
+
+                if configure.elapsed().unwrap().as_secs() > 24 * 60 * 60
+                    || last_scan.elapsed().as_secs() < 2
+                // For the first day after installation, only debounce to 2s
+                {
+                    info!(context, "Not scanning, less than a minute elapsed");
+                    return Ok(());
+                }
+            }
+        }
+        info!(context, "Starting full folder scan");
+        last_scan.replace(Instant::now());
+
+        self.setup_handle(context).await?;
+        let session = self.session.as_mut();
+        let session = session.context("scan_folders(): IMAP No Connection established")?;
+        let folders: Vec<_> = session.list(Some(""), Some("*")).await?.collect().await;
+
+        for folder in folders {
+            // TODO Maybe exclude folders that are watched anyway
```

Comment:
Maybe exclude folders that are watched anyway

Actually I don't think we should do this now as we will probably introduce a heuristics which folders to watch, so that we won't have 3 configs but a Vec (or something) of watched folders.